martin/src/pg/configurator.rs

```diff
@@ -232,12 +234,54 @@ impl PgBuilder {
     }
 }
 
+/// Try to find any ID column in a list of table columns (properties) that match one of the given `id_column` values.
+/// If found, modify `id_column` value on the table info.
+fn update_id_column(id: &str, inf: &mut TableInfo, auto_tables: &PgBuilderAuto) {
+    let Some(props) = inf.properties.as_mut() else { return };
+    let Some(try_columns) = &auto_tables.id_columns else { return };
+
+    for key in try_columns.iter() {
+        let (column, typ) = if let Some(typ) = props.get(key) {
+            (key, typ)
+        } else {
+            match find_kv_ignore_case(props, key) {
+                Ok(Some(result)) => {
+                    info!("For source {id} from table {}.{}, feature ID '{key}' was not found, but found '{result}' instead.", inf.schema, inf.table);
+                    (result, props.get(result).unwrap())
+                }
+                Ok(None) => continue,
+                Err(multiple) => {
+                    error!("Unable to set feature ID from table {}.{} because `{key}` has no exact match and more than one potential matches: {}", inf.schema, inf.table, multiple.join(", "));
+                    continue;
+                }
+            }
+        };
+        // ID column can be any integer type as defined in
+        // https://github.com/postgis/postgis/blob/559c95d85564fb74fa9e3b7eafb74851810610da/postgis/mvt.c#L387C4-L387C66
+        if typ != "int4" && typ != "int8" && typ != "int2" {
```

Comment:
need to double check if PG can ever return `int` or some other types here?  Likely not, just a thought.

Reply:
According to : https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-INT We have covered for all integer types.